src/rviz/CMakeLists.txt

```diff
@@ -145,6 +145,8 @@ target_link_libraries(${PROJECT_NAME}
   yaml-cpp
 )
 
+add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
```

Comment:
I think this might need to be `add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})`, since you're generating messages and using them in the same package, see:

https://docs.ros.org/lunar/api/catkin/html/howto/format2/building_msgs.html (bottom of the page)

Also see this for the origin of my above recommended syntax:

https://wiki.ros.org/catkin/CMakeLists.txt#Important_Prerequisites.2BAC8-Constraints (last bullet in section)